.github/workflows/ci.yml

```diff
@@ -176,6 +176,8 @@ jobs:
       - uses: ./.github/actions/upload
         with:
           test-report-retention-days: ${{ env.TEST_REPORT_RETENTION_DAYS }}
+          # only upload heap dumps if running on forks, where no secrets are used
+          upload-heap-dump: ${{ secrets.SECRETS_PRESENT == '' && github.repository_owner != 'trinodb' }}
```

Comment:
Can you test that the secret indeed matches empty string when it's missing? For example by adding a `run: echo ${{ secrets.SECRETS_PRESENT }}: boolean=${{ secrets.SECRETS_PRESENT == '' }}`?

(Not sure if valid syntax)

Reply:
in the worst case we won't get any heap dumps.

@nineinchnick can we verify this thing works by adding some OOM in the test (maybe clone this PR + change)?
or we need to have this merged into master first?

Reply:
Of course, I had to rerun the pipeline like 5 times, but finally, it works:
![image](https://user-images.githubusercontent.com/795177/203733545-59aa1f8c-ea33-46b1-adcb-5176a1e3d21c.png)